Tests/PINRemoteImageTests.m

```diff
@@ -1418,6 +1468,28 @@ - (void)testRetry
   method_exchangeImplementations(originalMethod, swizzledMethod);
 }
 
+
+- (void)testDownloadTaskWhenReDownload
+{
+    self.semaphore = dispatch_semaphore_create(0);
+    self.imageManager = [[PINRemoteImageManager alloc] init];
+    self.imageManager.sessionManager.delegate = self;
+    
+    PINRemoteImageTask *firstTask = nil;
+    NSUUID *firstUUID = [self.imageManager downloadImageWithURL:[self transparentPNGURL] options:0 progressDownload:nil completion:nil];
+    dispatch_semaphore_wait(self.semaphore, [self timeout]);
+    firstTask = self.firstDownloadTask;
+
+    [self.imageManager cancelTaskWithUUID:firstUUID];
+    
+    PINRemoteImageTask *secondTask = nil;
+    [self.imageManager downloadImageWithURL:[self transparentPNGURL] options:0 progressDownload:nil completion:nil];
+    dispatch_semaphore_wait(self.semaphore, [self timeout]);
+    secondTask = self.secondDownloadTask;
+    
+    XCTAssert(firstTask != secondTask);
```

Comment:
So I think this test will work but it involves caching the task, and not the key/request. So this test would not compile let alone fail w/o your change. Is it possible to add a test that will compile w/o the fix and fail to pass, and compile w/ the fix and pass the test? Thanks.

Reply:
Emm, sorry my English is poor, I may don't understand it.

Reply:
@bolsinga I think testing this will be exceedingly difficult without a testing network stack we can control. For now, let's merge this and think about how we can create a mocked networking stack since it would make *all* of our tests much easier to write.

Reply:
ok